src/index.tsx

```diff
@@ -195,21 +195,28 @@ export default function AxeMode({ children, disabled }: AxeModeProps) {
     }
 
     if (childrenRef.current) {
-      if (idleId) {
+      if (idleId && 'cancelIdleCallback' in window) {
         // requestIdleCallback has no types:
         // https://github.com/microsoft/TypeScript/issues/21309
         // @ts-ignore
         cancelIdleCallback(idleId);
         setIdleId(null);
       }
 
-      // requestIdleCallback has no types:
-      // https://github.com/microsoft/TypeScript/issues/21309
-      // @ts-ignore
-      const id = requestIdleCallback(() => {
+      function getViolations() {
         validateNode(childrenRef.current as ElementContext).then(setViolations);
-      });
-      setIdleId(id);
+      }
+
+      // Safari does not support requestIdleCallback 😔
+      if ('requestIdleCallback' in window) {
+        // requestIdleCallback has no types:
+        // https://github.com/microsoft/TypeScript/issues/21309
+        // @ts-ignore
+        const id = requestIdleCallback(getViolations);
+        setIdleId(id);
+      } else {
+        getViolations();
+      }
     }
   }, [children, disabled]);
```

Comment:
This won't reliably update when the DOM updates, just when `children` changes. The DOM could be updated by a queued async task, for example and our component may not capture violations in the updated node tree. I might remove the `children` dependency and use a `MutationObserver` to react to actual DOM changes

Reply:
That's true. It also wouldn't catch any imperative DOM mutations done in effects, right? Such as the ones we do to place the overlays.

Maybe let's test this out in real apps and do it as a follow-up? 🙂